components/raftstore/src/store/peer.rs

```diff
@@ -3391,13 +3382,41 @@ where
             return;
         }
 
-        // Reverse sort peers based on `next_idx`, then try to get a potential quorum.
-        next_idxs.sort_by(|x, y| y.1.cmp(&x.1));
-        let (mut potential_quorum, mut quorum_ok) = (HashSet::default(), false);
-        for &(peer_id, _, usage) in &next_idxs {
-            if matches!(usage, Some(DiskUsage::AlreadyFull)) {
-                continue;
+        // Reverse sort peers based on `next_idx` and `usage`, then try to get a potential quorum.
+        next_idxs.sort_by(|x, y| {
```

Comment:
Nice try but the code is ugly. You can map tuple.3 to an integer, and then comparing them with such simple rules:
1. check disk usage first;
2. then check `next_idx` if necessary